workspace/__ardulib__/ST7735/XST7735.cpp

```diff
@@ -0,0 +1,200 @@
+
+#include "XST7735.h"
+
+inline uint16_t swapColor565(uint16_t color) {
+    return (color << 11) | (color & 0x07E0) | (color >> 11);
+}
+
+ST7735::ST7735(uint8_t cs, uint8_t dc, uint8_t rst, uint8_t type) {
+    _cs = cs;
+    _rs = dc;
+    _rst = rst;
+    _type = type;
+    _width = ST7735_TFTWIDTH;
```

Comment:
Probably the implementation of the current library works fine only with the displays 128x160 (and in this case, I prefer to name it `ST7735_128x160`), I did not dive deep in the commands.
But it looks like `_width` and `_height` are hardcoded, but these hardcoded sizes do not used anywhere in the code. If it's the universal implementation for most of the ST7735 displays, I propose to remove `ST7735_TFTWIDTH` and `...HEIGHT` and get these sizes from the constructor arguments.

P.S.
Google says that there are ST7735 displays of other sizes, such as 128x128.

Reply:
It is hard to guarantee the error-free work (e.g the proper byte sequences for initialization) of display of this kind as we don't have real hardware to test on.

I suggest to leave it as it is. Let's take a look. If some user responds about an unsupported 7735 display it will be a great reason to expand the lib with a height/width constructor.